pkg/controller/cainjector/setup.go

```diff
@@ -108,7 +116,13 @@ func RegisterCertificateBased(mgr ctrl.Manager) error {
 	}
 	for _, setup := range injectorSetups {
 		if err := Register(mgr, setup, sources...); err != nil {
-			return err
+			if meta.IsNoMatchError(err) {
```

Comment:
So I see three options for improving this...

1) add a new `--injector-targets` flag that allows specifying which injectors to run
2) make use of a 'feature flag' to toggle this specific injector on/off (i.e. `AuditRegistrationInjector=true`)
3) add a new `IsAlpha` method to the Injector interface, and suppress this log/change this log messages *iff* the feature is marked alpha and fails to start with this error.

I feel like (2) or (3) is probably easiest to implement - (3) means we can have this enabled out the box for users that have a cluster with this API enabled.

Reply:
I will go with (3).